packages/web/app/components/Invoice/EditInvoiceModal/InvoiceItem.jsx

```diff
@@ -345,17 +357,34 @@ export const InvoiceItemRow = ({
             </ViewOnlyCell>
           )}
         </StyledItemCell>
-        <StyledItemCell width="10%" sx={{ flexGrow: 1 }} paddingLeft="10px">
+        <StyledItemCell width="11%" sx={{ flexGrow: 1 }} paddingLeft="10px">
           <PriceCell $hasLargeFont={!editable}>
-            <PriceText $isCrossedOut={!!discountPrice}>{price}</PriceText>
-            {!!discountPrice && (
-              <ThemedTooltip
-                key={item.discount?.reason}
-                title={item.discount?.reason}
-                open={item.discount?.reason ? undefined : false}
-              >
-                <span>{discountPrice}</span>
-              </ThemedTooltip>
+            {hidePriceInput ? (
+              <>
+                <PriceText $isCrossedOut={!!discountPrice}>{price}</PriceText>
+                {!!discountPrice && (
+                  <ThemedTooltip
+                    key={item.discount?.reason}
+                    title={item.discount?.reason}
+                    open={item.discount?.reason ? undefined : false}
+                  >
+                    <span>{discountPrice}</span>
+                  </ThemedTooltip>
+                )}
+              </>
+            ) : (
+              item.productId && (
+                <Field
+                  name={`invoiceItems.${index}.productPrice`}
+                  component={NumberField}
+                  min={0}
+                  max={999999}
+                  onInput={validateDecimalPlaces}
+                  size="small"
+                  required
+                  style={{ width: '100%' }}
+                />
```

Comment:
```suggestion

const PriceField = ({ field, ...props }) => {
  const handleChange = e => {
    let { value } = e.target;
    // If value is empty string and negative sign just return
    if (value === '' || /^[−-]/.test(value)) {
      field.onChange(value);
      return;
    }

    // IF the value is a number, set it to 2 decimal places
    let num = parseFloat(value);
    if (!isNaN(num)) {
      field.onChange(num.toFixed(2));
    }
  };

  return <NumberField value={field.value} onInput={handleChange} step="0.01" {...props} />;
};

                <Field
                  name={`invoiceItems.${index}.productPrice`}
                  component={PriceField}
                  min={0}
                  max={999999}
                  onInput={validateDecimalPlaces}
                  size="small"
                  required
                  style={{ width: '100%' }}
                />
```

It might be better to make a custom field time that extends Number field. Something like this (I haven't tested this code) because I think that setting e.target.value could have unexpected consequences.

Reply:
I've tried this, but `field.onChange` didn't directly prevent typing invalid number when input
I've implement the `PriceField` component btw